airflow/models/dag.py

```diff
@@ -3314,7 +3316,14 @@ def bulk_write_to_db(
                     input_datasets[DatasetModel.from_public(dataset)] = None
             curr_outlet_references = curr_orm_dag and curr_orm_dag.task_outlet_dataset_references
             for task in dag.tasks:
-                dataset_outlets = [x for x in task.outlets or [] if isinstance(x, Dataset)]
+                dataset_outlets: list[Dataset] = []
+                dataset_alias_outlets: list[DatasetAlias] = []
+                for outlet in task.outlets:
```

Comment:
hi @Lee-W -- we just starting testing our upgrade for `2.9.1` to `2.10.4` and spent a while trying to debug a `'NoneType' object is not iterable` error that was showing up when serializing our dags. We had a function that created tasks that defaulted the outlet kwarg as `None` instead of `[]`:
```python
def make_task(task_name,  outlets = None)
```
Updating the default kwarg value to the empty list fixed it but just wanted to mention that it might be helpful to provide the default iterable empty list like the previous version did:
```python
for outlet in task.outlets or []
```

Reply:
I think Airflow is not expecting `outlets` to be used this way.

https://github.com/apache/airflow/blob/c083e456fa02c6cb32cdbe0c9ed3c3b2380beccd/airflow/models/baseoperator.py#L1103-L1121

@uranusjr WDYT?